### PR TITLE
Fix model_json and nitweb

### PR DESCRIPTION
**Pull request by Morriar**
Was broken since #2312 

* Migrate `model_json` to new serialization API
* Add tests to `model_json` to ensure this doesn't happen anymore
* Migrate `web` package and `nitweb` to new API

@xymus a lot of serialization for you. Note: for each entity, I needed two deserialization version `full` and no full. Wasn't sure on how to handle this concern properly with your API

### CODE REVIEW

File: src/model/model_json.nit [hunk 1]
```diff
@@ -36,296 +36,239 @@ class MEntityRef
...
+	serialize
+
+	redef fun core_serialize_to(v) do
+		v.serialize_attribute("name", name)
+		v.serialize_attribute("class_name", class_name)
+		v.serialize_attribute("full_name", full_name)
+		v.serialize_attribute("mdoc", mdoc_or_fallback)
+		v.serialize_attribute("visibility", visibility.to_s)
+		v.serialize_attribute("modifiers", collect_modifiers)
+		v.serialize_attribute("location", location)
```
Comment (xymus) [inline src/model/model_json.nit]: This is a nice clean up!

**Comment by Morriar** [comment]
Reroll with @xymus suggestions on the two serialization versions

File: src/model/model_json.nit [hunk 2]
```diff
@@ -26,7 +26,7 @@ module model_json
 
 import model::model_collect
 import json::static
-import json
+intrude import json::serialization_write
```
Comment (xymus) [inline src/model/model_json.nit]: Do you still need the `intrude`?

**Comment by Morriar** [comment]
test this please

**Pull request merged**